tasks/AccelerationController.cpp

```diff
@@ -31,10 +31,35 @@ bool AccelerationController::configureHook()
     cmdVector = Eigen::VectorXd::Zero(numberOfThrusters);
     expectedEffortVector = Eigen::VectorXd::Zero(6);
     names = _names.get();
+    base::MatrixXd weighingMatrix;
+    base::VectorXd thrustersWeights = _thrusters_weights.get();
+
+    if(thrustersWeights.size() != numberOfThrusters)
+        return false;
```

Comment:
One more: I would accept an empty weight vector (in which case one would weight all thrusters the same), and require the vector to be empty if _svd_calculation is false (since you don't take weighting into account in this case)